src/action.js

```diff
@@ -46,30 +49,31 @@ async function action() {
 
     const client = github.getOctokit(core.getInput("token"));
 
-    if (debugMode) core.info(`reportPaths: ${reportPaths}`);
-    const reportsJsonAsync = getJsonReports(reportPaths);
-    const changedFiles = await getChangedFiles(base, head, client);
-    if (debugMode) core.info(`changedFiles: ${debug(changedFiles)}`);
-
-    const reportsJson = await reportsJsonAsync;
-    if (debugMode) core.info(`report value: ${debug(reportsJson)}`);
-    const reports = reportsJson.map((report) => report["report"]);
-
-    const overallCoverage = process.getOverallCoverage(reports);
-    if (debugMode) core.info(`overallCoverage: ${overallCoverage}`);
+    const coverageInfo = await getCoverageInfo(reportPaths, base, head, client, debugMode);
+    const overallCoverage = coverageInfo.overallCoverage;
+    const filesCoverage = coverageInfo.filesCoverage;
     core.setOutput(
       "coverage-overall",
       parseFloat(overallCoverage.project.toFixed(2))
```

Comment:
Would you happen to know why it is parseFloat'ing a float-to-string conversion? 

Like if the number was `.1`, it'd become .10 after toFixed and back to `.1` after the parseFloat.